### PR TITLE
Add IOCTL to get next free minor

### PR DESCRIPTION
**Pull request by crawfxrd**
The agent (any user, really) currently gets a free minor by parsing the proc file JSON and iterates through the existing devices. Push this complexity down into the driver, where it's a simple for-loop.

### CODE REVIEW

**Comment by nixomose** [comment]
my unsolicited 2 cents.
the only reason to get the next minor is to find a spot for the snapshot to take up.
rather than make a function that somebody can call to get the next minor, why not just have the driver figure it out when it goes to take a snapshot, since it's the atomic authority on what's available. I'm forgetting the details, but there's a bit of a window where two callers could get the same next minor if they tried to snap at the same time.
why not make the driver manage the whole thing while you're doing all this. then taking a snap would [somehow] return the minor that it allcoated.
just a thought.

**Comment by crawfxrd** [comment]
That's the eventual goal. I'm first looking to use the [IDR](https://www.kernel.org/doc/html/latest/core-api/idr.html) for managing minors, using [loop](https://elixir.bootlin.com/linux/v5.2/source/drivers/block/loop.c#L2093) and [dm](https://elixir.bootlin.com/linux/v5.2/source/drivers/md/dm.c#L1858) as a reference.

File: src/dattobd.c [hunk 1]
```diff
@@ -4245,8 +4245,20 @@ static int ioctl_dattobd_info(struct dattobd_info *info){
...
+static int get_free_minor(void)
+{
+	struct snap_device *dev;
+	int i;
+
+	tracer_for_each_full(dev, i){
+		if(!dev) return i;
+	}
+
+	return -ENODEV;
```
Comment (tcaputi) [inline src/dattobd.c]: for consistency, use ENOENT here

**Comment by Conan-Kudo** [comment]
Merged, thanks!

**Pull request merged**